packages/medusa/src/api/routes/admin/products/create-variant.ts

```diff
@@ -103,6 +119,48 @@ import { validator } from "../../../../utils/validator"
  *   "500":
  *     $ref: "#/components/responses/500_error"
  */
+
+enum actions {
+  createVariant = "createVariant",
+  createInventoryItem = "createInventoryItem",
+  attachInventoryItem = "attachInventoryItem",
+}
+
+const simpleFlow: TransactionStepsDefinition = {
+  next: {
+    action: actions.createVariant,
+    maxRetries: 0,
+  },
+}
+
+const flowWithInventory: TransactionStepsDefinition = {
+  next: {
+    action: actions.createVariant,
+    forwardResponse: true,
+    maxRetries: 0,
+    next: {
+      action: actions.createInventoryItem,
+      forwardResponse: true,
+      maxRetries: 0,
+      next: {
+        action: actions.attachInventoryItem,
+        noCompensation: true,
+        maxRetries: 0,
+      },
+    },
+  },
+}
```

Comment:
thought (non-blocking): I think we should do an iteration on the TransactionStepsDefinition that makes it easier to "inject" steps into the definition. 

Initially, this would benefit us by making it straightforward to go from `simpleFlow` to `flowWithInventory` without having to write two full definitions. But in the future we could imagine also making it easier for external developers to customize add their own additional steps at various points in the flow.